models/repos.js

```diff
@@ -1,109 +1,205 @@
-      update: update
+      ready: multidatReady,
+      update: updateDats
     },
     effects: {
-      open: openDat,
-      share: shareDat,
       create: createDat,
-      delete: deleteDat,
-      download: downloadDat,
-      'add-dir': addDirectory,
-      deleteConfirm: deleteConfirmed
+      remove: removeDat,
+      open: openDirectory,
+      share: shareDat,
+      clone: cloneDat
     }
   }
 
-  function handleManagerUpdate (send, done) {
-    manager.on('update', debounce(() => {
-      send('repos:update', manager.get(), done)
-    }), 1500, true)
+  // boot multidat, create the ~/Downloads/dat directory
+  function startMultidat (send, done) {
+    var dblocation = path.join(process.env.HOME)
+    var dbfile = path.join(dblocation, '.dat-desktop-db.json')
+
+    var arr = [
+      function (next) {
+        mkdirp(dblocation, next)
+      },
+      function (val, next) {
+        var db = toilet(dbfile)
+        Multidat(db, next)
+      },
+      function (val, next) {
+        send('repos:ready', function (err) {
+          if (err) return next(err)
+          next(null, val)
+        })
+      }
+    ]
+
+    waterfall(arr, function (err, multidat) {
+      if (err) return done(err)
+      manager = createManager(multidat, function (err, dats) {
+        if (err) return done(err)
+        send('repos:update', dats, done)
+      })
+    })
+  }
+
+  // signal multidat is ready to accept values
+  function multidatReady (state, data) {
+    return { ready: true }
   }
 
-  function update (state, data) {
-    return {
-      updateIndex: state.updateIndex + 1,
-      values: data
-    }
+  // update the values with then new values from the manager
+  function updateDats (state, data) {
+    return { values: data }
   }
 
-  function openDat (state, dat) {
-    open(dat.dir, (err) => {
-      if (err) throw err
+  // handle IPC events from the server
+  function handleIpc (send, done) {
+    ipc.on('link', (event, url) => {
+      const key = encoding.decode(url)
+      send('repos:clone', key, done)
     })
+    ipc.on('log', (ev, str) => console.log(str))
+    ipc.send('ready')
   }
 
+  // open the dat archive in the native filesystem explorer
+  function openDirectory (state, data, send, done) {
+    assert.ok(data.path, 'repos-model.openDirectory: data.path should exist')
+    open(data.path, done)
+  }
+
+  // choose a directory and convert it to a dat archive
   function createDat (state, data, send, done) {
-    const files = dialog.showOpenDialog({
-      properties: ['openDirectory']
+    var pathname = data
+    if (!pathname) {
+      var files = dialog.showOpenDialog({
+        properties: ['openDirectory']
+      })
+      if (!files || !files.length) return done()
+      pathname = files[0]
+    }
+    manager.create(pathname, done)
+  }
+
+  function removeDat (state, data, send, done) {
+    assert.ok(data.key, 'repos-model.deleteDat: data.key should exist')
+
+    if (data.confirmed === true) {
```

Comment:
because the case where `data.confirmed === false` always happens first, can we list it first here too?